test/jdk/javax/imageio/plugins/jpeg/JpegWriterWriteNonOpaqueIndexColorModelTest.java

```diff
@@ -0,0 +1,96 @@
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8351108
+ * @summary This test verifies that attempting to write a JPEG using
+ *          ImageIO.write(..) fails by returning `false` for
+ *          translucent IndexColorModels.
+ */
+
+import javax.imageio.ImageIO;
+import java.awt.Transparency;
+import java.awt.image.BufferedImage;
+import java.awt.image.IndexColorModel;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+public class JpegWriterWriteNonOpaqueIndexColorModelTest {
+    public static void main(String[] args) throws IOException {
+        boolean b1 = testJpegWriter(Transparency.OPAQUE, "OPAQUE", true);
+        boolean b2 = testJpegWriter(Transparency.BITMASK, "BITMASK", false);
+        boolean b3 = testJpegWriter(Transparency.TRANSLUCENT, "TRANSLUCENT", false);
+        if (!(b1 && b2 && b3))
+            throw new Error("Test failed");
+    }
+
+    /**
+     * Test writing a JPEG using a IndexColorModel
+     *
+     * @param imageType a Transparency constant (OPAQUE, BITMASK, TRANSLUCENT) that is used to
+     *                  define a new IndexColorModel.
+     * @param name the name of the Transparency constant
+     * @param expectedWriteReturnValue the expected return value of <code>ImageIO.write(..)</code>
+     * @return true if <code>ImageIO.write(..) == expectedWriteReturnValue</code>, or false otherwise.
+     *         This also returns false if <code>ImageIO.write(..)</code> throws an exception. We do not
+     *         expect that to happen anymore. (Prior to resolving 8351108 ImageIO threw an exception.)
+     */
+    private static boolean testJpegWriter(int imageType, String name, boolean expectedWriteReturnValue) {
+        System.out.println();
+        System.out.println("TESTING " + name);
+        try {
+            byte[] gray = new byte[256];
+            for (int a = 0; a < gray.length; a++) {
+                gray[a] = (byte) a;
+            }
+
+            byte[] alpha = new byte[256];
+            if (imageType == Transparency.OPAQUE || imageType == Transparency.BITMASK) {
+                for (int a = 0; a < alpha.length; a++) {
+                    alpha[a] = -1;
+                }
+                if (imageType == Transparency.BITMASK) {
+                    alpha[0] = 0;
+                }
+            } else if (imageType == Transparency.TRANSLUCENT) {
+                for (int a = 0; a < alpha.length; a++) {
+                    alpha[a] = (byte) a;
+                }
+            }
+            IndexColorModel indexColorModel = new IndexColorModel(8, 256, gray, gray, gray, alpha);
```

Comment:
```suggestion
            }

            IndexColorModel indexColorModel = new IndexColorModel(8, 256, gray, gray, gray, alpha);
```
I think a blank line here will visually separate the block of preparing the arrays for `IndexColorModel` from their usage.

Creating an instance of `IndexColorModel` marks the start of the *real test*.